pkg/diagnostics/client/run_diagnostics_pod.go

```diff
@@ -0,0 +1,163 @@
+package client
+
+import (
+	"bufio"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+const (
+	DiagnosticPodName = "DiagnosticPod"
+)
+
+// DiagnosticPod is a diagnostic that runs a diagnostic pod and relays the results.
+type DiagnosticPod struct {
+	KubeClient          kclient.Client
+	Namespace           string
+	Level               int
+	Factory             *osclientcmd.Factory
+	PreventModification bool
+	ImageTemplate       variable.ImageTemplate
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d *DiagnosticPod) Name() string {
+	return DiagnosticPodName
+}
+
+// Description is part of the Diagnostic interface and provides a user-focused description of what the diagnostic does.
+func (d *DiagnosticPod) Description() string {
+	return "Create a pod to run diagnostics from the application standpoint"
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d *DiagnosticPod) CanRun() (bool, error) {
+	if d.PreventModification {
+		return false, fmt.Errorf("Running the diagnostic pod is an API change, which is prevented as you indicated.")
```

Comment:
i disagree that these should be treated like framework errors but it does not matter enough to me to argue about it.